evm/vm/base.py

```diff
@@ -210,6 +210,9 @@ def state_in_temp_block(self):
         temp_block = self.generate_block_from_parent_header_and_coinbase(header, header.coinbase)
         prev_hashes = (header.hash, ) + self.previous_hashes
         state = self.get_state(block_header=temp_block.header, prev_hashes=prev_hashes)
+        if state.gas_used > 0:
+            raise AssertionError("There must not be any gas used in a fresh temporary block")
```

Comment:
👎 assertions in runtime code.  If this is supposed to be an unreachable code path you can just use `Exception`.